### DISCUSSION TITLE
relation behavior

### DISCUSSION THREAD

**Discussion by as7ali** [wintercms/winter#1035] [Q&A] [answered]
I have a problem in relation behavior in backend controller 
I used belew codes but when reload backend page get the error 500 in browser and 15 same error in event log with message  

>    Symfony\Component\ErrorHandler\Error\FatalError: Uncaught BadMethodCallException: Call to undefined method System\Models\Parameter::hasDatabaseTable() in C:\xampp\htdocs\mywinter\vendor\laravel\framework\src\Illuminate\Support\Traits\ForwardsCalls.php:71  

folder/models
     ***Category.php***
  ` public $table = 'as7_relationtest_category';   // table fields : id,name `
  ` public $hasMany = [   'products' => [Product::class] ];`
    ***Product.php***
`public $table = 'as7_relationtest_product'; ';   // table fields : id,name,category_id`
`public $belongsTo = [   'category' => [Category::class]    ];`

folder/models/category
***fields.yaml***
`fields:`
    `  name:`
       `    label: Name`
      `    type: text`
`tabs:`
    `  fields:`
        `    products:`
           `      type: partial`
            `      path: $/as7/relationtest/controllers/category/_products.htm`
            `      tab: 'Tab 1'`
            
folder/models/product
***columns.yaml***
`columns:`
   `  id:`
        `    label: id`
        `    type: number`
    `  name:`
        `    label: name`
        `    type: text`
***fields.yaml***
`fields:`
    `  name:`
       `    label: Name`
        `    type: text`

folder/controllers
***Category.php***
`use Backend\Classes\Controller;`
`use BackendMenu;`
`class Category extends Controller`
`{`
` public $implement = [        'Backend\Behaviors\ListController',        'Backend\Behaviors\FormController'  ,'\Backend\Behaviors\RelationController'  ];`
` `
    `public $listConfig = 'config_list.yaml';`
   `public $formConfig = 'config_form.yaml';`
   `public $relationConfig = 'config_relation.yaml';`
    `public function __construct()    {        parent::__construct();`
            ` BackendMenu::setContext('As7.Relationtest', 'main-menu-item');    }}`

folder/controllers/category
***config_relation.yaml***
`products:`
  `  label: products`
  `  manage:`
  `      form: $/as7/relationtest/models/product/fields.yaml `
  `      list: $/as7/relationtest/models/product/columns.yaml `
 `   view: `
  `      list: $/as7/relationtest/models/product/columns.yaml `
   `      toolbarButtons: `

**Comment by WebVPF** [comment, marked as answer]
Syntax error in folder/models/Product.php

```php
public $table = 'as7_relationtest_product'; ';   // table fields : id,name,category_id
public $belongsTo = [   'category' => [Category::class]    ];
```

Twice `'; ';` in `t_product'; ';   // table`

Corrected:

```php
public $table = 'as7_relationtest_product'; // table fields : id,name,category_id

public $belongsTo = [
    'category' => [Category::class]
];
```


In folder/controllers/Category.php values in `$implement`

```php
use Backend\Classes\Controller;
use BackendMenu;

class Category extends Controller
{
    public $implement = [
        \Backend\Behaviors\FormController::class,
        \Backend\Behaviors\ListController::class,
        \Backend\Behaviors\RelationController::class,
    ];

    public $listConfig = 'config_list.yaml';
    public $formConfig = 'config_form.yaml';
    public $relationConfig = 'config_relation.yaml';

    public function __construct()
    {
        parent::__construct();
        BackendMenu::setContext('As7.Relationtest', 'main-menu-item');
    }
}
```
